src/Symfony/Component/Translation/Translator.php

```diff
@@ -72,32 +72,45 @@ class Translator implements TranslatorInterface, TranslatorBagInterface
     private $configCacheFactory;
 
     /**
+     * @var TranslatorBagInterface
+     */
+    private $translatorBag;
+
+    /**
      * Constructor.
      *
-     * @param string               $locale   The locale
-     * @param MessageSelector|null $selector The message selector for pluralization
-     * @param string|null          $cacheDir The directory to use for the cache
-     * @param bool                 $debug    Use cache in debug mode ?
+     * @param string                 $locale        The locale
+     * @param MessageSelector|null   $selector      The message selector for pluralization
+     * @param string|null            $cacheDir      The directory to use for the cache
+     *                                              Deprecated since version 2.8, to be removed in 3.0.
+     * @param bool                   $debug         Use cache in debug mode ?
+     *                                              Deprecated since version 2.8, to be removed in 3.0.
+     * @param TranslatorBagInterface $translatorBag
      *
      * @throws \InvalidArgumentException If a locale contains invalid characters
      *
      * @api
      */
-    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false)
+    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false, TranslatorBagInterface $translatorBag = null)
```

Comment:
Ok then maybe something like:

```php
public function __construct($locale, $translatorBag, $selector = null, $cacheDir = null, $debug = false)
{
    $this->setLocale($locale);

    if ($translatorBag instanceof TranslatorBagInterface) {
        $this->translatorBag = $translatorBag;
        $this->selector = $selector ?: new MessageSelector();
    } else {
        @trigger_error('The '.__CLASS__.' constructor will require a TranslatorBagInterface for second parameter since 3.0.', E_USER_DEPRECATED);

        // Parameters are shifted of one offset
        $this->selector = $translatorBag ?: new MessageSelector();
        $this->cacheDir = $selector;
        $this->debug = $cacheDir;
    }
}
```